jib-core/src/main/java/com/google/cloud/tools/jib/frontend/MainClassFinder.java

```diff
@@ -157,8 +184,9 @@ public static String resolveMainClass(
                     && Modifier.isPublic(main.getModifiers())) {
                   classNames.add(fileClass.getName());
                 }
-              } catch (NoSuchMethodException ignored) {
+              } catch (NoSuchMethodException | NoClassDefFoundError ignored) {
```

Comment:
Shouldn't this not be happening anymore if it's taken care of in `findClass`?

Reply:
I mostly just added it for safety in case there are edge cases we haven't considered.